src/libfetchers/tree-info.hh

```diff
@@ -11,7 +11,7 @@ namespace nix::fetchers {
 
 struct TreeInfo
 {
-    Hash narHash;
+    std::optional<Hash> narHash;
```

Comment:
Another

Reply:
Though I think this one is fine and just means the hash is calculted---obviously these builtins aren't going to be making non-CA paths.

Reply:
I think this one should not be optional. Anyway it's better to leave libfetchers unchanged since it's all rewritten in the flakes branch.

Reply:
We thought it might be easy to fix this, but it isn't either: the caching stuff doesn't store the `narHash` and since the cache also doesn't distinguish between stores, doing a `queryPathInfo` to get the Nar hash in the case of a cache-hit doesn't work.

Perhaps something could be made to work, but like you said `libfetchers` is all rewritten anyways, and I think the 2 lines we changed in `libfetchers` with the `std::optional` is much more minimal than whatever it would take to make it work without the `std::optional`.

Reply:
This change did go away with the flakes branch being merged, yay!